src/components/WalletInit/ConnectNanoX/SaveNanoXScreen.js

```diff
@@ -136,8 +136,8 @@ export default injectIntl(
           walletImplementationId,
           hwDeviceInfo,
         )
-        navigation.navigate(ROOT_ROUTES.WALLET)
         saveHW(hwDeviceInfo)
+        navigation.navigate(ROOT_ROUTES.WALLET)
```

Comment:
it makes more sense to save the device info before navigating but technically I think this doesn't make any difference, so the issue doesn't come from here.